subgraphs/aave-forks/protocols/aave-amm/src/mapping.ts

```diff
@@ -350,21 +288,77 @@ export function handleRepay(event: Repay): void {
     event,
     event.params.amount,
     event.params.reserve,
-    getProtocolData(),
+    protocolData,
     event.params.user
   );
 }
 
 export function handleLiquidationCall(event: LiquidationCall): void {
+  // when liquidator chooses to receive AToken (event.params.receiveAToken==true),
+  // the transfer of AToken will trigger a BalanceTransfer event and the
+  // handleCollateralTransfer event will transfer the position from the liquidatee
+  // to the liquidator
+  const createLiquidatorPosition = false;
+
   _handleLiquidate(
     event,
     event.params.liquidatedCollateralAmount,
     event.params.collateralAsset,
-    getProtocolData(),
+    protocolData,
     event.params.liquidator,
     event.params.user,
     event.params.debtAsset,
-    event.params.debtToCover
+    event.params.debtToCover,
+    createLiquidatorPosition
+  );
+}
+
+export function handleFlashloan(event: FlashLoan): void {
+  const flashloanPremium = getOrCreateFlashloanPremium(protocolData);
+  flashloanPremium.premiumRateTotal = FLASHLOAN_PREMIUM_TOTAL;
+  flashloanPremium.save();
+
+  _handleFlashLoan(
+    event.params.asset,
+    event.params.amount,
+    event.params.initiator,
+    protocolData,
+    event,
+    event.params.premium,
+    flashloanPremium
+  );
+}
+
+export function handleSwapBorrowRateMode(event: Swap): void {
+  const interestRateMode = event.params.rateMode.toI32();
+  if (
+    ![InterestRateMode.STABLE, InterestRateMode.VARIABLE].includes(
+      interestRateMode
+    )
+  ) {
+    return;
```

Comment:
probably want to throw an error log here. This would warrant code changes if aave introduced another mode

Reply:
same as above?